sdk/batch/Microsoft.Azure.Management.Batch/src/Generated/Models/ContainerRegistry.cs

```diff
@@ -30,16 +29,20 @@ public ContainerRegistry()
         /// <summary>
         /// Initializes a new instance of the ContainerRegistry class.
         /// </summary>
+        /// <param name="registryServer">The registry URL.</param>
         /// <param name="userName">The user name to log into the registry
         /// server.</param>
         /// <param name="password">The password to log into the registry
         /// server.</param>
-        /// <param name="registryServer">The registry URL.</param>
-        public ContainerRegistry(string userName, string password, string registryServer = default(string))
+        /// <param name="identityReference">The reference to the user assigned
+        /// identity to use to access an Azure Container Registry instead of
+        /// username and password.</param>
+        public ContainerRegistry(string registryServer = default(string), string userName = default(string), string password = default(string), ComputeNodeIdentityReference identityReference = default(ComputeNodeIdentityReference))
```

Comment:
hmmm, since these parameters are all strings, this looks like a breaking change we cannot mitigate.

Reply:
I think the easiest way to deal with this may be to add the x-ms-external directive to this class in the C# autorest readme - this would allow you to provide this class is customizations, with a non-breaking parameter ordering.  If we don't want to make this change, or something similar, we will have to request a breaking change exception.